src/UNL/UCBCN/Manager/Calendar.php

```diff
@@ -83,15 +83,24 @@ public function getEvents()
     private function archiveEvents() {
         # find all posted (upcoming) events on the calendar
         $events = $this->calendar->getEvents('posted');
+        $archived_events = $this->calendar->getEvents('archived');
 
         # check each event to see if it has passed
         foreach ($events as $event) {
             # we will consider it passed if all datetimes are in the past
             $datetimes = $event->getDatetimes();
-            $archive = true;
+            $archive = TRUE;
             foreach ($datetimes as $datetime) {
+                $recurring_dates = $datetime->getAllDates();
+                foreach($recurring_dates as $recurring_date) {
+                    if ($recurring_date->recurringdate >= date('Y-m-d')) {
```

Comment:
Hmmm, I think `$recurring_date->recurringdate` is in the same format as `date('Y-m-d')`, so they should be comparable. However, the safer solution would be to `strtotime` them both, but that also adds a tiny bit of overhead.